apps/aesophia/src/aeso_icode_to_asm.erl

```diff
@@ -945,6 +943,14 @@ push_label(Label) -> {push_label, Label}.
 jump(Label)    -> [push_label(Label), i(?JUMP)].
 jump_if(Label) -> [push_label(Label), i(?JUMPI)].
 
+exhaust_gas() ->
```

Comment:
There is actually one instruction that is dedicated as "the" invalid instruction 0xfe INVALID.
Perhaps not in scope of this ticket but perhaps we should use that here.

Reply:
> There is actually one instruction that is dedicated as "the" invalid instruction 0xfe INVALID.

I had started coding that (i.e. using "the" invalid instruction) though it was going to impact more the semantics of the Sophia compiled bytecode and - consequently - the tests (match on `out_of_gas`). Hence I settled on exhausting gas at the Sophia bytecode level - as I saw it was used in Sophia compiled bytecode already for switch-case not matching any case.

> Perhaps not in scope of this ticket but perhaps we should use that here.

If we decide to go that way, I am happy to take that forward though I would prefer to keep it as distinct ticket/task/PR.